### ISSUE TITLE
FAB: Allow content instead of menu

### ISSUE THREAD

**Issue by testing138** [primefaces-extensions/primefaces-extensions.github.com#741]
Hi all, 

It would be awesome if you can create a widget as it´s in use for many famous systems (zendesk, intercom). You can use it for chat, chat bot, knowledge articles etc.

Please find examples:
- Widget is in the right corner:
![image](https://user-images.githubusercontent.com/52133436/68151688-748f8d00-ff42-11e9-8645-d639b65877d2.png)

- User can click on it and a new panel is opening:
![image](https://user-images.githubusercontent.com/52133436/68151740-912bc500-ff42-11e9-8379-04373eaa6ab8.png)

Panel with scrollbar:
![image](https://user-images.githubusercontent.com/52133436/68151857-ca643500-ff42-11e9-9a43-5f65da3a20f7.png)

**Events:**
- If as user click on it
- If as user close it 

**Attributes:**
- Open on start (yes /no)
- iconInit (String)
- iconClose (String)
- autoStartPageLoad (if panel should be open on page load)
- attributes like for floatingActionButton

It seems it´s a combination between:
pe:FloatingActionButton and pe:slideout

**Comment by tandraschko** [comment]
looks like a button with some styling + overlayPanel.
-1 for add something that is more like a composite component.

**Comment by melloware** [comment]
Community discussion:  https://forum.primefaces.org/viewtopic.php?f=14&t=60376

@tandraschko We already have Floating Action Button that assumes you want to use it as a menu. I think it could easily be modified to take panel content as well....similar to this Jquery component
https://www.jqueryscript.net/menu/Material-style-Floating-Button-Panel-Plugin-For-jQuery-st-action-panel.html

**Comment by melloware** [comment]
Or maybe we just need to add "on" handlers to FAB so a user can open an Overlay Panel from FAB?

**Comment by jepsar** [comment]
I could throw in an Ajax event

**Comment by melloware** [comment]
Yeah I think open and close ajax events would be good.  Then we could use "pe:javascript" to open and close a OverlayPanel.

**Comment by melloware** [comment]
After you do it I will add a showcase example doing exactly like ZenDesk.

**Comment by testing138** [comment]
Additional to that - sometimes it make sense to have the p:menuitem and if you click on one of the <p:menuitem, it will open a panel as in my first screenshot.

E.g. you have one button:
- p:menuitem no1 will open chat panel
- p:menuitem no2 will open knowledge panel
- p:menuitem no3 will open email panel

But if you could implement such like mentioned in my first post, that would be very great.

`<pe:fab icon="fa fa-link" source="mypanel">
<p:outputPanel id="mypanel">
// my chat window etc.
</p:outputPanel>  
</pe:fab>`

**Comment by testing138** [comment]
And one more wishes:

**1) Something like badget, like:** 
Attributes: 
a) showValues = true / false
b) valuesNumber = int number
![image](https://user-images.githubusercontent.com/52133436/68188140-c9fe8500-ffa8-11e9-9e30-11119bfa3273.png)

**2) Overlay Panel is open on loading screen**
Attributes: 
a) loadOnStart = true / false
-> As soon page is loaded, the panel is open automatically.

**3) Open Panel from Bean / Close from Bean
The idea is e.g. if a user is 2 minutes on a specific page, the panel widget should oben automatically. 
E.g. if the user is on the "price page" you can send a "push notification" which open the panel automatically with a chat panel and a text like "Do you have any question" appears.

**Comment by melloware** [comment]
Please see https://primefaces.github.io/primefaces/7_1/#/components/overlaypanel?id=overlaypanel OverLayPanel documentation it already does most of what you want. because it has 3 client side JS methods.  So loading the panel on start simply call `PF('overlayPanel').show();`  So none of that will be necessary for this change.

**Comment by rider87** [comment]
This enhancement would be very nice.
I agree to melloware: 2) and 3) can be done with OverlayPanl, but not 1). 
This would be also great to integrate it here

**Comment by melloware** [comment]
This is fixed for 7.1.
To acheive the example above...
```xml
<pe:fab id="btnFab" icon="fa fa-question" iconActive="fa fa-times" style="background: green;">
	<pe:javascript event="open" execute="PF('overlayWidget').show('btnFab');"/> 
	<pe:javascript event="close" execute="PF('overlayWidget').hide();"/> 
</pe:fab>
```

Closed:
![image](https://user-images.githubusercontent.com/4399574/68211641-efd65a80-ffa5-11e9-92c9-731d9e2750f9.png)

Opened:
![image](https://user-images.githubusercontent.com/4399574/68211681-011f6700-ffa6-11e9-8166-659f34a3f139.png)

**Comment by melloware** [comment]
@testing138 To Achive iOS badge functionality see the new component: https://github.com/primefaces-extensions/primefaces-extensions.github.com/issues/724

You can probably steal the CSS from our source code to apply the badge to almost anything.

**Comment by rider87** [comment]
> 
> 
> This is fixed for 7.1.
> To acheive the example above...
> 
> ```
> <pe:fab id="btnFab" icon="fa fa-question" iconActive="fa fa-times" style="background: green;">
> 	<pe:javascript event="open" execute="PF('overlayWidget').show('btnFab');"/> 
> 	<pe:javascript event="close" execute="PF('overlayWidget').hide();"/> 
> </pe:fab>
> ```
> 
> Closed:
> ![image](https://user-images.githubusercontent.com/4399574/68211641-efd65a80-ffa5-11e9-92c9-731d9e2750f9.png)
> 
> Opened:
> ![image](https://user-images.githubusercontent.com/4399574/68211681-011f6700-ffa6-11e9-8166-659f34a3f139.png)

means it can be used now? Or at least in version 7.1 ?
If yes, when will 7.1 be available?

**Comment by melloware** [comment]
7.1 will be based on PF 7.1 which is due to have its first Release Candidate deployed in the next couple of weeks.

If you need this in 7.0.3 i would just grab the 7.0.3 release https://github.com/primefaces-extensions/core/releases

Make the changes I made in this commit: https://github.com/primefaces-extensions/core/commit/59906bbf037654682bc88b531684eafa0ad58351

And do a `mvn clean install` to build yourself a 7.0.4 version.
